tcms/rpc/api/testexecution.py

```diff
@@ -170,6 +170,12 @@ def update(execution_id, values, **kwargs):
         if form.cleaned_data['tested_by']:
             tcr.tested_by = form.cleaned_data['tested_by']
 
+        if form.cleaned_data['case_text_version']:
+            tcr.case_text_version = form.cleaned_data['case_text_version']
+            latest_version = tcr.case.history.latest().history_id
+            if tcr.case_text_version != latest_version:
+                tcr.case_text_version = latest_version
```

Comment:
this is wrong. if I do `TestExecution.update executionId {'case_text_version': 2}`. I would expect to end up with `2` for `case_text_version`, regardless of what the latest it. This is not the case now.

there are two options here:
1. have an API that sets the `case_text_version` of an execution to the latest (We could use `TestExecution.update`, but the would should be something like `set_case_text_version_to_latest` and be a Boolean field). This way you would call it like: `jsonRPC('TestExecution.update', [executionId, {'set_case_text_version_to_latest': true}])` and achieve the desired result
2. have a standard update API which sets the values that you passed to it (like this, without the last 3 lines). This way the client would be responsible for passing the latest version of `case_text_version`. In our case this would mean that we need to define another API that gives you the latest `case_text_version` of a `TestCase`.

you can choose one of them, or come up with another one. IMO version 1) would be easier to implement as involves less coding

Reply:
Instead of `{'set_case_text_version_to_latest': true}` - why not special case the `case_text_version: latest` value? It's the same field but it can receive a special value. Django already has the machinery to easily handle this and you already have exposure to that.

Reply:
yes, I like that solution, because it handles both cases